lib/origen/registers/bit_collection.rb

```diff
@@ -332,7 +351,11 @@ def write(value, options = {})
         value = value.data if value.respond_to?('data')
 
         size.times do |i|
-          self[i].write(value[i], options)
+          if @with_bit_order == :msb0
+            self[i].write(value[size - i - 1], options)
```

Comment:
This update was necessary because when with_bit_order == :msb0 the [] method interprets the bit index as an msb0 index. I will need to review the other bit collection methods to see if similar updates are needed and will likely update the with_lsb0 method to accept a code block. If a block is given the interpretation will be changed to lsb0 during the code block execution and restored after.